server/controllers/stock/requisition/requisition.js

```diff
@@ -155,7 +199,85 @@ exports.update = async (req, res, next) => {
       delete requisition.uuid;
     }
 
-    requisition.date = requisition.date ? new Date(requisition.date) : new Date();
+    if (requisition.movementRequisition) {
+      const dataMovementRequisition = db.convert(
+        req.body.movementRequisition, ['stock_requisition_uuid', 'document_uuid'],
+      );
+
+      delete requisition.movementRequisition;
+
+      // Just to make a link between the stock issues coming from the requisition
+      await db.exec('INSERT INTO stock_requisition_movement SET ?;', dataMovementRequisition);
+
+      const checkRequisitionBalance = `
+      SELECT COUNT(balance.inventory_uuid) AS numberInventoryPartial
+        FROM (
+        SELECT req.inventory_uuid, req.code, req.inventoryType,
+        (req.quantity - IF(mouv.quantity, mouv.quantity, 0)) AS quantity
+            FROM (
+              SELECT BUID(i.uuid) inventory_uuid, i.code, it.text as inventoryType, sri.quantity
+              FROM stock_requisition_item sri
+              JOIN inventory i ON i.uuid = sri.inventory_uuid
+              JOIN inventory_type it ON i.type_id = it.id
+              WHERE sri.requisition_uuid = ?
+            ) AS req
+            LEFT JOIN (
+              SELECT BUID(inv.uuid) AS inventory_uuid, inv.code, inv.text AS inventoryType, SUM(m.quantity) AS quantity
+              FROM stock_movement AS m
+              JOIN lot AS l ON l.uuid = m.lot_uuid
+              JOIN inventory AS inv ON inv.uuid = l.inventory_uuid
+              JOIN stock_requisition_movement AS srm ON srm.document_uuid = m.document_uuid
+              WHERE srm.stock_requisition_uuid = ?
+              GROUP BY inv.uuid
+            ) AS mouv ON mouv.inventory_uuid = req.inventory_uuid
+          ) AS balance
+          WHERE balance.quantity > 0;
+        `;
+
+      const movementStatus = await db.one(checkRequisitionBalance, [
+        dataMovementRequisition.stock_requisition_uuid, dataMovementRequisition.stock_requisition_uuid,
+      ]);
+
+      const checkExcessiveBalance = `
+      SELECT COUNT(balance.inventory_uuid) AS numberExcessive
+        FROM (
+        SELECT req.inventory_uuid, req.code, req.inventoryType,
+        (req.quantity - IF(mouv.quantity, mouv.quantity, 0)) AS quantity
+            FROM (
+              SELECT BUID(i.uuid) inventory_uuid, i.code, it.text as inventoryType, sri.quantity
+              FROM stock_requisition_item sri
+              JOIN inventory i ON i.uuid = sri.inventory_uuid
+              JOIN inventory_type it ON i.type_id = it.id
```

Comment:
You don't need the JOINs on the inventory_type.  IN fact, if all you need is the movement quantity, you probably don't need a lot of these JOINs.  Can you try to remove them?